File: java/client/src/test/java/org/signal/libsignal/net/UnauthCredentialsServiceTest.kt
//
// Copyright 2026 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

package org.signal.libsignal.net

import kotlinx.coroutines.test.runTest
import org.signal.libsignal.internal.NativeTestingNice
import kotlin.test.Test
import kotlin.test.assertEquals
import kotlin.test.assertIs

class UnauthCredentialsServiceTest {
  @Test
  fun testCheckSvrCredentials() {
    runTest {
      GrpcTestCase.runTests(
        NativeTestingNice.TESTING_CheckSvrCredentialsTests(),
        { tokio, listener ->
          UnauthenticatedChatConnection.fakeConnect(tokio, listener, Network.Environment.STAGING)
        },
        ::UnauthCredentialsService,
        invoke = { chat, req ->
          chat.checkSvrCredentials(
            number = req.number,
            credentials = req.passwords,
          )
        },
        check = { expected, actual ->
          assertEquals(expected.toMap(), assertIs<RequestResult.Success<Map<String, AuthCheckResult>>>(actual).result)
        },
      )
    }
  }
}

File: node/ts/test/chat/UnauthCredentialsServiceTest.ts
//
// Copyright 2026 Signal Messenger, LLC.
// SPDX-License-Identifier: AGPL-3.0-only
//

import { config, expect, use } from 'chai';
import chaiAsPromised from 'chai-as-promised';

import * as NativeNice from '../../NativeNice.js';
import * as util from '../util.js';
import { UnauthCredentialsService } from '../../net.js';
import { connectUnauth, defineTestGrpcCases } from './ServiceTestUtils.js';

use(chaiAsPromised);

util.initLogger();
config.truncateThreshold = 0;

describe('UnauthCredentialsService', () => {
  describe('checkSvrCredentials', () => {
    defineTestGrpcCases(
      NativeNice.TESTING_CheckSvrCredentialsTests(),
      connectUnauth<UnauthCredentialsService>,
      async (
        chat: UnauthCredentialsService,
        { number, passwords }: NativeNice.CheckSvrCredentialsArgs,
        resp: Array<[string, NativeNice.AuthCheckResult]>
      ) => {
        const out = await chat.checkSvrCredentials({
          number,
          credentials: passwords,
        });
        expect(out).to.deep.equal(new Map(resp));
      }
    );
  });
});
